File: Array/Hard/4-median-of-two-sorted-arrays.py
# ═══════════════════════════════════════════════════════
# Problem: 4. Median of Two Sorted Arrays
# Difficulty: Hard
# Topics: Array, Binary Search, Divide and Conquer
# Runtime: 11 ms (Beats 8.6%)
# Memory: 12.6 MB (Beats 51.8%)
# Submitted: Sep 2, 2026
# Link: https://leetcode.com/problems/median-of-two-sorted-arrays/
# ═══════════════════════════════════════════════════════

class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        list1=[]
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        for i in range(0,len(nums1)):
            list1.append(nums1[i])

        for i in range(0,len(nums2)):
            list1.append(nums2[i])

        n=len(list1)
        list1.sort()

        if(n%2!=0):
            return float(list1[n//2])
        else:
            return(list1[(n//2)-1]+list1[n//2])/2.0
    

File: Array/Hard/4-median-of-two-sorted-arrays.cpp
// ═══════════════════════════════════════════════════════
// Problem: 4. Median of Two Sorted Arrays
// Difficulty: Hard
// Topics: Array, Binary Search, Divide and Conquer
// Runtime: 4 ms (Beats 18.9%)
// Memory: 95.9 MB (Beats 25.5%)
// Submitted: Sep 2, 2026
// Link: https://leetcode.com/problems/median-of-two-sorted-arrays/
// ═══════════════════════════════════════════════════════

class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int>result;
        for(int i=0;i<nums1.size();i++){
            result.push_back(nums1[i]);

        }
        for(int i=0;i<nums2.size();i++){
            result.push_back(nums2[i]);
        }
         int n=result.size();

         sort(result.begin(),result.end());

        if(n%2!=0){
            return result[n/2];
        }
        else{
            return (result[(n / 2) - 1] + result[n / 2]) / 2.0;
        }
    }
};
